test/test_weighted_mean.py

```diff
@@ -0,0 +1,147 @@
+import os
+from pathlib import Path
+
+import pytest
+
+import dask.array as da
+import numpy as np
+import numpy.testing as nt
+
+import uxarray as ux
+
+current_path = Path(os.path.dirname(os.path.realpath(__file__)))
+
+
+csne30_grid_path = current_path / 'meshfiles' / "ugrid" / "outCSne30" / "outCSne30.ug"
+csne30_data_path = current_path / 'meshfiles' / "ugrid" / "outCSne30" / "outCSne30_vortex.nc"
+
+quad_hex_grid_path = current_path / 'meshfiles' / "ugrid" / "quad-hexagon" / "grid.nc"
+quad_hex_data_path_face_centered = current_path / 'meshfiles' / "ugrid" / "quad-hexagon" / "data.nc"
+quad_hex_data_path_edge_centered = current_path / 'meshfiles' / "ugrid" / "quad-hexagon" / "random-edge-data.nc"
+
+
+def test_quad_hex_face_centered():
+    """Compares the weighted average computation for the quad hexagon grid
+    using a face centered data variable to the expected value computed by
+    hand."""
+    uxds = ux.open_dataset(quad_hex_grid_path, quad_hex_data_path_face_centered)
+
+    # expected weighted average computed by hand
+    expected_weighted_mean = 297.55
```

Comment:
Can you compute this within Python? I think if you can avoid hard coding in the answer that is ideal, although I know this can't always be avoided.

Reply:
+1 this! Even if it is inevitable to use constants, they shouldn't be showing up as magic numbers within the code and instead should go into some kind of test constants